rust-runtime/aws-smithy-types/src/timeout/http.rs

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use crate::tristate::TriState;
+use std::time::Duration;
+
+/// HTTP timeouts used by `DynConnector`s
+#[non_exhaustive]
+#[derive(Clone, PartialEq, Default, Debug)]
+pub struct Http {
+    /// A limit on the amount of time after making an initial connect attempt on a socket to complete the connect-handshake.
+    connect: TriState<Duration>,
+    write: TriState<Duration>,
+    /// A limit on the amount of time an application takes to attempt to read the first byte over an
+    /// established, open connection after write request. This is also known as the
+    /// "time to first byte" timeout.
+    read: TriState<Duration>,
+    /// A limit on the amount of time a TLS handshake takes from when the `CLIENT HELLO` message is
+    /// sent to the time the client and server have fully negotiated ciphers and exchanged keys.
+    tls_negotiation: TriState<Duration>,
```

Comment:
Same comment about docs on the private fields.

Reply:
good callout